File: VB/XafSolution.Module.Win/Controllers/RoleGeneratorControllerWin.vb
Imports System
Imports System.Collections.Generic
Imports System.Configuration
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports XafSolution.Module.Controllers

Namespace XafSolution.Module.Win.Controllers
    Public Class RoleGeneratorControllerWin
        Inherits RoleGeneratorController

        Protected Overrides Sub SaveFile(ByVal updaterCode As String)
            Dim saveFileDialog As New SaveFileDialog()
            saveFileDialog.Filter = "vb|*.vb"
            If saveFileDialog.ShowDialog() = DialogResult.OK Then
                Using file As New StreamWriter(saveFileDialog.FileName, False)
                    file.Write(updaterCode)
                End Using
            End If
        End Sub
        Protected Overrides Function IsEnableRoleGeneratorAction() As Boolean
            Dim enableRoleGeneratorActionString As String = ConfigurationManager.AppSettings("EnableRoleGeneratorAction")
            Return If(enableRoleGeneratorActionString Is Nothing, False, Boolean.Parse(enableRoleGeneratorActionString))
        End Function
    End Class
End Namespace

File: CS/XafSolution.Module.Win/Controllers/RoleGeneratorControllerWin.cs
using System;
using System.IO;
using System.Linq;
using System.Windows.Forms;
using XafSolution.Module.Controllers;

namespace XafSolution.Module.Win.Controllers {
    public class RoleGeneratorControllerWin : RoleGeneratorController {
        protected override void SaveFile(string updaterCode) {
            SaveFileDialog saveFileDialog = new SaveFileDialog();
            saveFileDialog.Filter = "cs|*.cs";
            if(saveFileDialog.ShowDialog() == DialogResult.OK) {
                using(StreamWriter file = new StreamWriter(saveFileDialog.FileName, false)) {
                    file.Write(updaterCode);
                }
            }
        }
    }
}
